### PR TITLE
[WB-6294] Warning when offline and resume

### PR DESCRIPTION
**Pull request by kptkin**
https://wandb.atlassian.net/browse/WB-6294

Description
-----------

What does the PR do?

When users are in offline mode, they shouldn't set the `resume` attribute. 
This PR will have a warning message indicating that resume will be ignored.

Testing
-------

How was this PR tested?

Add new tests to capture the warning when offline and different values for resume

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------

Proper warning message for the behavior of offline with resume

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/sdk/wandb_init.py [hunk 1]
```diff
@@ -485,6 +496,12 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
             with telemetry.context(run=run) as tel:
                 tel.feature.offline = True
             run_proto = backend.interface._make_run(run)
+
+            if s.resume is not None:
+                self._error_out(
+                    backend, error_message="Offline mode does not support resume"
```
Comment (raubitsj) [inline wandb/sdk/wandb_init.py:502]: If we did the check before we created the backend, we wouldnt have to do the backend.cleanup() teardown code.
(that code is fraught with cases it doesn't handle properly)

Comment (kptkin) [reply wandb/sdk/wandb_init.py:502]: yeah i was trying to figure out if I should do it as soon as possible. 
Seems like the answer is yes. Let me move it up

**Review by raubitsj — APPROVED** [review]
Great!

File: tests/wandb_run_test.py [hunk 2]
```diff
@@ -163,3 +163,26 @@ def test_except_hook(test_settings):
...
+
+
+@pytest.mark.parametrize(
+    "resume, found",
+    [
+        ("auto", True),
+        ("allow", True),
+        ("never", True),
+        ("must", True),
+        ("", True),
```
Comment (raubitsj) [inline tests/wandb_run_test.py:180]: empty string warns?  this is unexpected isnt it?

Comment (kptkin) [reply tests/wandb_run_test.py:180]: yep, seems to be due to[ this piece of code](https://github.com/wandb/client/blob/master/wandb/sdk/wandb_settings.py#L1057-L1063).

**Pull request merged**